probe-rs/src/architecture/xtensa/xdm.rs

```diff
@@ -519,78 +496,61 @@ impl Xdm {
     pub(super) fn resume(&mut self) -> Result<(), XtensaError> {
         tracing::debug!("resuming...");
         // Clear pending interrupts first that would re-enter us into the Stopped state
-        self.write_nexus_register({
+        self.schedule_write_nexus_register({
             let mut clear_status = DebugStatus(0);
 
             clear_status.set_debug_pend_host(true);
             clear_status.set_debug_pend_break(true);
 
             clear_status
-        })?;
-
-        self.execute_instruction(Instruction::Rfdo(0))?;
+        });
 
-        Ok(())
+        self.schedule_execute_instruction(Instruction::Rfdo(0));
+        self.execute()
     }
 
-    pub fn write_instruction(&mut self, instruction: Instruction) -> Result<(), XtensaError> {
+    pub(super) fn schedule_write_instruction(&mut self, instruction: Instruction) {
         tracing::debug!("Preparing instruction: {:?}", instruction);
         self.last_instruction = Some(instruction);
 
         match instruction.encode() {
             InstructionEncoding::Narrow(inst) => {
-                self.write_nexus_register(DebugInstructionRegister(inst))
+                self.schedule_write_nexus_register(DebugInstructionRegister(inst));
             }
         }
     }
 
-    pub fn execute_instruction(&mut self, instruction: Instruction) -> Result<(), XtensaError> {
+    pub(super) fn schedule_execute_instruction(&mut self, instruction: Instruction) {
         tracing::debug!("Executing instruction: {:?}", instruction);
         self.last_instruction = Some(instruction);
 
         match instruction.encode() {
             InstructionEncoding::Narrow(inst) => {
-                self.write_nexus_register(DebugInstructionAndExecRegister(inst))?;
+                self.schedule_write_nexus_register(DebugInstructionAndExecRegister(inst));
             }
         }
-
-        self.wait_for_exec_done()
```

Comment:
This wait increases the amount of data transferred during memory read/write considerably.